eth-contracts/contracts/service/ServiceProviderFactory.sol

```diff
@@ -272,4 +312,60 @@ contract ServiceProviderFactory is RegistryContract {
             _endpoint
         );
     }
+
+    function isValidServiceType(bytes32 _serviceType)
+    external view returns (bool isValid) 
+    {
+      for (uint i = 0; i < validServiceTypes.length; i ++) {
+        if (validServiceTypes[i] == _serviceType) {
+          return true;
+        }
+      }
+      return false;
+    }
+
+    function getValidServiceTypes()
+    external view returns (bytes32[] memory types) 
+    {
+      return validServiceTypes;
+    }
+
+    function getServiceStakeInfo(bytes32 _serviceType) 
+    external view returns (uint min, uint max)
+    {
+      return (serviceTypeStakeRequirements[_serviceType].minStake, serviceTypeStakeRequirements[_serviceType].maxStake);
+    }
+
+    function getAccountStakeBounds(address sp)
+    external view returns (uint min, uint max) 
+    {
+      uint minStake = 0;
+      uint maxStake = 0;
+      uint validTypesLength = validServiceTypes.length;
+      for (uint i = 0; i < validTypesLength; i++) {
+        bytes32 serviceType = validServiceTypes[i];
+        (uint typeMin, uint typeMax) = this.getServiceStakeInfo(serviceType); 
+        uint numberOfEndpoints = this.getServiceProviderIdsFromAddress(sp, serviceType).length; 
+        minStake += (typeMin * numberOfEndpoints); 
+        maxStake += (typeMax * numberOfEndpoints); 
+      }
+      return (minStake, maxStake);
+    }
+
+    function validateAccountStakeBalances(address sp) internal view
```

Comment:
// validate that the service provider is between the min and max stakes for all their registered services